File: LeetCode_2588/Main.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    long long beautifulSubarrays(vector<int>& nums) {
        unordered_map<long long, long long> count;
        long long xorVal = 0, subArray = 0;
        count[0] = 1;

        for (long long num : nums) {
            xorVal ^= num;
            subArray += count[xorVal];
            count[xorVal]++;
        }
        return subArray;
    }
};

File: LeetCode_2588/Main.py
import math
class Solution(object):
    def beautifulSubarrays(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        from collections import defaultdict
        cnt = defaultdict(int)
        cnt[0] = 1
        xor = 0
        subArray = 0
        for i in nums:
            xor ^= i
            if cnt[xor] > 0:
                subArray += cnt[xor]
            cnt[xor] += 1
        return subArray
